instrumentation/opentelemetry-instrumentation-flask/src/opentelemetry/instrumentation/flask/__init__.py

```diff
@@ -188,14 +196,18 @@ def _instrument(self, **kwargs):
         self._original_flask = flask.Flask
```

Comment:
What if user instruments with `instrument()`? We should provide this functionality for that too.

Reply:
The current implementation is using `get_default_span_name`. I am wondering how do we hook the name call back in this case?

Reply:
Similar to [requests](https://github.com/open-telemetry/opentelemetry-python-contrib/pull/158/files), use a kwarg.

Reply:
I added it. One thing is we are patching the `flask.Flask` to get `instrument()` to work but if an import of Flask is made before the `FlaskInstrumentor().instrument()` then it doesn't do anything. I don't think if this is documented somewhere.

Reply:
Yes this is known behaviour.